projects/graph/graph.py

```diff
@@ -11,45 +11,145 @@ def add_vertex(self, vertex):
+        #  Create an empty set to store visited nodes
+        #  Crea3te an empty Queue an enqueue the starting vertex
+        #  While the Queue is not empty...
+            # Dequeue the first vertex
+            #  If that vertex has not been visited...
+                #  Mark it ad visited
+                #  Then add all of its neighbors to the back of the queue
+
     def dft(self, starting_vertex):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
         """
-        pass  # TODO
-    def dft_recursive(self, starting_vertex):
+        print('\n\nDFT:\n')
+        visited = set()
+        s = Stack()
+        s.push(starting_vertex)
+        while s.size() > 0:
+            v = s.pop()
+            if v not in visited:
+                visited.add(v)
+                print(v)
+                for neighbor in self.vertices[v]:
+                    s.push(neighbor)
+    def dft_recursive(self, starting_vertex, visited=set()):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
         This should be done using recursion.
         """
-        pass  # TODO
+        if len(visited) < 1:
+            print('\n\nDFT Recursive:\n')
+        s = Stack()
+        s.push(starting_vertex)
+        while s.size() > 0:
+            v = s.pop()
+            if v not in visited:
+                visited.add(v)
+                print(v)
+                for neighbor in self.vertices[v]:
+                    self.dft_recursive(neighbor, visited)
+
     def bfs(self, starting_vertex, destination_vertex):
         """
         Return a list containing the shortest path from
         starting_vertex to destination_vertex in
         breath-first order.
         """
-        pass  # TODO
+        print('\n\nBFS:\n')
+        #  Create an empty Queue and enqueue A PATH TO the starting vertex
+        q = Queue()
+        #  Create an empty set to store visited nodes
+        visited = set()
+        q.enqueue([starting_vertex])
+
+        #  While the queue is not empty..
+        while q.size() > 0:
+            #  Dequeue the first PATH
+            path = q.dequeue()
+            #  GRAB THE VERTEX FROM THE END OF THE PATH
+            v = path[-1]
+            #  IF VERTEX == TARGET...
+            if v == destination_vertex:
+                # RETURN PATH
+                return path
+            #  If that vertex has not been visited...
+            if v not in visited:
+                #  Mark it as visited
+                visited.add(v)
+                #  Then add A PATH TO ALL of its neighbors to the back of the queue
+                for neighbor in self.vertices[v]:
+                    #  Copy the path
+                    path_copy = list(path)
+                    #  Append neighbor to the back of the copy
+                    path_copy.append(neighbor)
+                    #  Enqueue copy
+                    q.enqueue(path_copy)
+
```

Comment:
looks good; would probably just move your destination check nested inside the v not in visited , prior to the add